synapse/config/emailconfig.py

```diff
@@ -294,8 +322,47 @@ def read_config(self, config, **kwargs):
                 if not os.path.isfile(p):
                     raise ConfigError("Unable to find email template file %s" % (p,))
 
+        subjects_config = email_config.get("subjects", {})
+        self.email_subjects = EmailSubjectConfig(
```

Comment:
I don't know if this would be better, but you could set the `default` value for each attribute on `EmailSubjectConfig` and then splat the config on top of it? It might give *terrible* error messages though

```python
@attr.s
class EmailSubjectConfig:
    message_from_person_in_room = attr.ib(type=str, default="[%(app)s] You have a message on %(app)s from %(person)s in the %(room)s room...")
    ...

self.email_subjects = EmailSubjectConfig(**subjects_config)
```

You could also use `setattr` with a loop, maybe...

Reply:
After discussing it in #synapse-dev, we decided to go for the solution implemented in https://github.com/matrix-org/synapse/pull/7846/commits/35a16fa048c45f24c6802b214fb2c871b531c058